.helm/ecamp3-logging/templates/elasticsearch.yaml

```diff
@@ -0,0 +1,143 @@
+              securityContext:
+                privileged: true
+                runAsUser: 0
+              command: [ 'sh', '-c', 'sysctl -w vm.max_map_count=262144' ]
+
+      volumeClaimTemplates:
+        - metadata:
+            name: elasticsearch-data
+            labels:
+              app: elasticsearch
+              {{- include "app.commonLabels" . | nindent 14 }}
+              {{- include "app.commonSelectorLabels" . | nindent 14 }}
+          spec:
+            accessModes:
+              - ReadWriteOnce
+            resources:
+              {{- toYaml .Values.elasticsearch.main.persistence.resources | nindent 14 }}
+            storageClassName: do-block-storage
+    - name: data
+      # this is here because we now know how to set it up
+      # if we need more nodes for our data, we can increase the count
+      count: {{ .Values.elasticsearch.data.replicas }}
+      config:
+        node.roles:
+          - "data"
+          - "ingest"
+          - "ml"
+          - "transform"
+      podTemplate:
+        spec:
+          containers:
+            - name: elasticsearch
+              resources:
+                 {{- toYaml .Values.elasticsearch.data.resources | nindent 16 }}
+              startupProbe:
+                exec:
+                  command:
+                    - bash
+                    - -c
+                    - /mnt/elastic-internal/scripts/readiness-probe-script.sh
+                failureThreshold: 20
+                initialDelaySeconds: 90
+                periodSeconds: 10
+                successThreshold: 1
+                timeoutSeconds: 12
+              readinessProbe:
+                exec:
+                  command:
+                    - bash
+                    - -c
+                    - /mnt/elastic-internal/scripts/readiness-probe-script.sh
+                failureThreshold: 3
+                initialDelaySeconds: 10
+                periodSeconds: 12
+                successThreshold: 1
+                timeoutSeconds: 12
+              env:
+                - name: READINESS_PROBE_TIMEOUT
+                  value: "20"
+                - name: ES_JAVA_OPTS
+                  value: "-Xms{{ $dataMinHeapSpace }}m -Xmx{{ $dataMaxHeapSpace }}m"
+          initContainers:
+            - name: sysctl
+              securityContext:
+                privileged: true
+                runAsUser: 0
+              command: [ 'sh', '-c', 'sysctl -w vm.max_map_count=262144' ]
+      volumeClaimTemplates:
+        - metadata:
+            name: elasticsearch-data
+            labels:
+              app: elasticsearch
+              {{- include "app.commonLabels" . | nindent 14 }}
+              {{- include "app.commonSelectorLabels" . | nindent 14 }}
+          spec:
+            accessModes:
+              - ReadWriteOnce
+            resources:
+              {{- toYaml .Values.elasticsearch.data.persistence.resources | nindent 14 }}
+            storageClassName: do-block-storage
```

Comment:
```
lucius@lucius-ThinkPad-P17-Gen-1:~$ helm search repo elastic
NAME                        	CHART VERSION	APP VERSION	DESCRIPTION                                       
bitnami/elasticsearch       	19.18.0      	8.12.1     	Elasticsearch is a distributed search and analy...
elastic/eck-elasticsearch   	0.9.1        	           	Elasticsearch managed by the ECK operator         
elastic/elasticsearch       	8.5.1        	8.5.1      	Official Elastic helm chart for Elasticsearch     
elastic/apm-attacher        	1.0.0        	           	A Helm chart installing the Elastic APM Kuberne...
elastic/apm-server          	8.5.1        	8.5.1      	Official Elastic helm chart for Elastic APM Server
elastic/eck-agent           	0.9.1        	           	Elastic Agent managed by the ECK operator         
elastic/eck-beats           	0.9.1        	           	Elastic Beats managed by the ECK operator         
elastic/eck-fleet-server    	0.9.1        	           	Elastic Fleet Server as an Agent managed by the...
elastic/eck-kibana          	0.9.1        	           	Kibana managed by the ECK operator                
elastic/eck-logstash        	0.9.1        	           	Logstash managed by the ECK operator              
elastic/eck-operator        	2.11.1       	2.11.1     	Elastic Cloud on Kubernetes (ECK) operator        
elastic/eck-operator-crds   	2.11.1       	2.11.1     	ECK operator Custom Resource Definitions          
elastic/eck-stack           	0.9.1        	           	Elastic Stack managed by the ECK Operator         
elastic/filebeat            	8.5.1        	8.5.1      	Official Elastic helm chart for Filebeat          
elastic/kibana              	8.5.1        	8.5.1      	Official Elastic helm chart for Kibana            
elastic/logstash            	8.5.1        	8.5.1      	Official Elastic helm chart for Logstash          
elastic/metricbeat          	8.5.1        	8.5.1      	Official Elastic helm chart for Metricbeat        
elastic/pf-host-agent       	8.12.1       	8.12.1     	Hyperscaler software efficiency. For everybody.   
elastic/profiling-collector 	8.12.1       	8.12.1     	Universal Profiling. Hyperscaler software effic...
elastic/profiling-symbolizer	8.12.1       	8.12.1     	Universal Profiling. Hyperscaler software effic...
bitnami/dataplatform-bp2    	12.0.5       	1.0.1      	DEPRECATED This Helm chart can be used for the ...
bitnami/kibana              	10.10.0      	8.12.1     	Kibana is an open source, browser based analyti...
```

As you see the eck-stack chart is stuck at version 0.9.1 while the operator is at 2.11.1
This may have happened at the same time as this commit:
https://github.com/elastic/cloud-on-k8s/commit/8dd27ce6afc2679117b74f64a297fa335ccbb80a

Where the docs state that
> The instructions below are intended to deploy the Helm charts from a local copy of this repository. Refer to https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-install-helm.html to install the charts from the official repository.

https://github.com/elastic/cloud-on-k8s/blob/main/deploy/README.md#eck-operator-and-eck-resources-helm-charts
The referenced link only mentions the operator chart.

which is more or less what i did.